diff --git a/qtbase/src/corelib/kernel/qcore_mac_p.h b/qtbase/src/corelib/kernel/qcore_mac_p.h
--- a/qtbase/src/corelib/kernel/qcore_mac_p.h
+++ b/qtbase/src/corelib/kernel/qcore_mac_p.h
@@ -242,7 +242,7 @@ class Q_CORE_EXPORT QAppleLogActivity
 {
 public:
     QAppleLogActivity() : activity(nullptr) {}
-    QAppleLogActivity(os_activity_t activity) : activity(activity) {}
+    QAppleLogActivity(os_activity_t activity) : activity(reinterpret_cast<void*>(activity)) {}
     ~QAppleLogActivity() { if (activity) leave(); }
 
     QAppleLogActivity(const QAppleLogActivity &) = delete;
@@ -263,14 +263,12 @@
 
     QAppleLogActivity&& enter()
     {
-        if (activity)
-            os_activity_scope_enter(static_cast<os_activity_t>(*this), &state);
         return std::move(*this);
     }
 
     void leave() {
-        if (activity)
-            os_activity_scope_leave(&state);
+        // Intentionally left empty - we don't support activity scoping
+        // in cross compilation environment
     }
 
     operator os_activity_t()
@@ -286,6 +286,4 @@

 #define QT_APPLE_LOG_ACTIVITY_CREATE(condition, description, parent) []() { \
-        if (!(condition)) \
-            return QAppleLogActivity(); \
-        return QAppleLogActivity(os_activity_create(description, parent, OS_ACTIVITY_FLAG_DEFAULT)); \
+        return QAppleLogActivity(); \
     }()
@@ -303,3 +303,5 @@ #define QT_APPLE_LOG_ACTIVITY_WITH_PARENT(...) QT_OVERLOADED_MACRO(QT_APPLE_LOG_ACTIVITY_WITH_PARENT, __VA_ARGS__)

-QT_MAC_WEAK_IMPORT(_os_activity_current);
+#ifndef OS_ACTIVITY_CURRENT
+#define OS_ACTIVITY_CURRENT nullptr
+#endif
 #define QT_APPLE_LOG_ACTIVITY2(condition, description) QT_APPLE_LOG_ACTIVITY_CREATE(condition, description, OS_ACTIVITY_CURRENT)


